### ISSUE TITLE
Re-scaling image patches reduces accuracy of the LPIPS perceptual metric

### ISSUE THREAD

**Issue by abhijay-g** [richzhang/PerceptualSimilarity#64]
I trained and tested LPIPS on the BAPPS dataset and the Validation dataset.

**2AFC scores on BAPPS**

LPIPS | Trained on | Tested on | Mean 2AFC
-- | -- | -- | --
alex | 64x64* | 64x64* | 69.83
 "|64x64* | 256x256 | 68.75
 "|256x256 | 64x64* | 65.88
 "|256x256 | 256x256 | 70.54

\* Here 64 means that the original BAPPS images of size 256 were rescaled to size 64 

**Comparing rescaled images via frequency spectrum in polar form**
Here we take crops from the 5220 images given in the validation set from the [CLIC - perceptual metric challenge](http://compression.cc/tasks/) and compute the difference between the frequency spectrums of a crop from a corrupted image vs its reference i.e. considering function F that calculates the frequency spectrum in polar form, the plots below are are 

> mean( abs ( F(X_0).crop() - F(X_ref).crop() ))

where X_0 is one of the corrupted images. 

In the fig below theleftmost plot is the mean calculated for crop patches of size 256, and on its right mean for crop patches of size 576 rescaled to size 256 using bicubic downsampling. Similarly, the crop patches of size 64 show a similar trend.

![fourierAnalysisOfBicubicScaling](https://user-images.githubusercontent.com/11029633/109372311-c1952200-785d-11eb-91ee-37753c291db9.png)
Note: Visualizing only one channel. Also, I wrote all this in a hurry, please let me know if you need more details.

It can be observed that some of the high and low-frequency information that could be important for a perceptual metric is lost during rescaling of the image patches.

**Hypothesis: The drop in accuracy stems from rescaling and not the scale of images.**
I tested the accuracy of the LPIPS models trained on image patches without rescaling

LPIPS | LPIPS trained on | Val data size | Val data scaled | Val Accuracy
-- | -- | -- | -- | --
alex | 256 | 986 |   | 0.767
vgg | 256 | 986 |   | 0.754
alex | 64 | 986 |   | 0.739
vgg | 64 | 986 |   | 0.745
alex | 256 | 986 | 256 | 0.678
vgg | 256 | 986 | 256 | 0.679
alex | 64 | 986 | 256 | 0.735
vgg | 64 | 986 | 256 | 0.74

I take an average of the gradients to compare image patches with no rescaling vs patches that are rescaled.
![multiple_grad_diff_no_scaling](https://user-images.githubusercontent.com/11029633/109372569-0cfc0000-785f-11eb-86d1-236685ebb42a.png)
![multiple_grad_diff_on_scaled](https://user-images.githubusercontent.com/11029633/109372572-0f5e5a00-785f-11eb-8f3a-bbc61f56121e.png)
Note: Visualizing the three channels rgb. Alex means LPIPS[Alex] here. The dataset used was the aforementioned validation set from CLIC challenge.

The issue does not arise due to the scale of the images but re-scaling of the images.
It was pointed out to me that this is common knowledge and there's nothing new here.

Since it is a perceptual metric the image patches should not be rescaled. I could be wrong on this and my guess is that I'm missing something important here. I would love to know your opinion regarding this issue.

P.S. An interesting argument raised by my advisor was that the original images must have also been re-scaled from a larger size to 256. I'm curious, then why do the results correlate with the other dataset (Val set from CLIC).

**Comment by richzhang** [comment]
Thanks for sharing the observations! That's quite interesting.

Yes, the images were originally 64x64 and upsampled to 256x256 (with nearest), just to display to Turkers. It's a bit of a mistake that they were downsampled down to bilinear back to 64x64

**Comment by abhijay-g** [comment]
Apologies for briefly removing the content. I thought I was being impolite by troubling you with something trivial. Thank you for your prompt reply. Your comment is really helpful. Thanks!
